src/http/client/client.cr

```diff
@@ -89,6 +89,25 @@ class HTTP::Client
     @compress = true
   end
 
+  # Creates a new HTTP client from a URI. Parses the *host*, *port*,
+  # and *ssl* configuration from the url provided. Port defaults to
+  # 80 if not specified unless using the https protocol, which defaults
+  # to the port to 443 and sets ssl to `true`.
+  #
+  # ```
+  # uri = URI.parse("https://secure.example.com")
+  # client = HTTP::Client.new(uri)
+  #
+  # client.ssl? # => true
+  # client.get("/")
+  # ```
+  def self.new(uri : URI)
+    is_ssl = uri.scheme == "https"
+    port = uri.port || (is_ssl ? 443 : 80)
+
+    new(uri.host.to_s, port.to_i, is_ssl)
```

Comment:
You can just pass `uri.port`, no need to do the `is_ssl ? 443 : 80` check, because that's done in the regular constructor (I removed it and specs still pass).